### PR TITLE
Updating settings to unify miniconf and encdec

### PR DESCRIPTION
**Pull request by ryan-summers**

### CODE REVIEW

**Review by jordens — COMMENTED** [review]
* wasn't the idea to get rid of either `BoosterMainBoardData` or `Properties`?
* I'm fine with removing the `IpAddr` wrapper and having the user do `set ip [1,2,3,4]` (add appropriate docs and help string)
* two clippy lints

**Comment by ryan-summers** [comment]
This PR replaces `Properties` with `BoosterMainBoardData`, so the structs are now unified. The diff makes that hard to see for some reason.

File: src/serial_terminal.rs [hunk 1]
```diff
@@ -312,6 +312,7 @@ fn handle_property_write(
         return;
     }
 
+    context.settings.properties = new_props;
```
Comment (ryan-summers) [inline src/serial_terminal.rs:315]: This fixes a bug in the last PR where the new settings weren't getting persisted properly.

**Review by jordens — APPROVED** [review]
All ACK, LGTM.

**Pull request merged**